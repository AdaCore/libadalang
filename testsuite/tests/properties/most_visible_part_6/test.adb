package body Test is
   function Constructor (Param : Integer) return Boolean is
   begin
      return Param > 42;
   end Constructor;
   --% p = node.f_subp_spec.p_params[0].p_defining_names[0]
   --% p.p_most_visible_part(p)
end Test;
