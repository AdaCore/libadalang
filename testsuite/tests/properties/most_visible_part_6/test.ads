package Test is
   function Constructor (Param : Integer) return Boolean;
   --% p = node.f_subp_spec.p_params[0].p_defining_names[0]
   --% p.p_most_visible_part(node)
   --% p.p_most_visible_part(node.p_body_part())
end Test;
