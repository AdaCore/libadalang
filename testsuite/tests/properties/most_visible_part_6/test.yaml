driver: inline-playground
input_sources: [test.ads, test.adb]
