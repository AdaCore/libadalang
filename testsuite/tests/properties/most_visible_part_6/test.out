Working on node <SubpBody ["Constructor"] test.adb:2:4-5:20>
============================================================

Set 'p' to 'node.f_subp_spec.p_params[0].p_defining_names[0]'
Result: <DefiningName "Param" test.adb:2:26-2:31>

Eval 'p.p_most_visible_part(p)'
Result: <DefiningName "Param" test.adb:2:26-2:31>

Working on node <SubpDecl ["Constructor"] test.ads:2:4-2:58>
============================================================

Set 'p' to 'node.f_subp_spec.p_params[0].p_defining_names[0]'
Result: <DefiningName "Param" test.ads:2:26-2:31>

Eval 'p.p_most_visible_part(node)'
Result: <DefiningName "Param" test.ads:2:26-2:31>

Eval 'p.p_most_visible_part(node.p_body_part())'
Result: <DefiningName "Param" test.adb:2:26-2:31>
